include/deal.II/lac/solver_idr.h

```diff
@@ -226,6 +226,54 @@ namespace internal
         }
       return *data[i];
     }
+
+
+
+    template <typename VectorType,
+              std::enable_if_t<!IsBlockVector<VectorType>::value, VectorType>
+                * = nullptr>
+    unsigned int
+    n_blocks(const VectorType &)
+    {
+      return 1;
+    }
+
+
+
+    template <typename VectorType,
+              std::enable_if_t<IsBlockVector<VectorType>::value, VectorType> * =
+                nullptr>
+    unsigned int
+    n_blocks(const VectorType &vector)
+    {
+      return vector.n_blocks();
+    }
+
+
+
+    template <typename VectorType,
+              std::enable_if_t<!IsBlockVector<VectorType>::value, VectorType>
+                * = nullptr>
+    VectorType &
+    block(VectorType &vector, const unsigned int b)
+    {
+      AssertDimension(b, 0);
+      (void)b;
+      return vector;
+    }
+
+
+
+    template <typename VectorType,
+              std::enable_if_t<IsBlockVector<VectorType>::value, VectorType> * =
+                nullptr>
+    typename VectorType::BlockType &
+    block(VectorType &vector, const unsigned int b)
+    {
+      AssertThrow(b == 1, ExcInternalError());
+      return vector.block(b);
+    }
```

Comment:
This code is the same as https://github.com/dealii/dealii/blob/2b2da6719b3cdea2fceea7e21a0cb510fd2b1fe3/include/deal.II/matrix_free/operators.h#L44-L91 - shouldn't we try to unify this e.g. in the file https://github.com/dealii/dealii/blob/master/include/deal.II/lac/block_vector_base.h ?